src/dasi/api/dasi_c.cc

```diff
@@ -179,6 +197,84 @@ int dasi_archive(dasi_t* dasi, const dasi_key_t* key, const void* data,
     });
 }
 
+int dasi_wipe(dasi_t* dasi, const dasi_query_t* query, const dasi_bool_t* doit, const dasi_bool_t* all,
+              dasi_wipe_t** wipe) {
+    return tryCatch([dasi, query, doit, all, wipe] {
+        ASSERT(dasi);
+        ASSERT(query);
+        ASSERT(doit);
+        ASSERT(all);
+        ASSERT(wipe);
+        *wipe = new dasi_wipe_t(dasi->wipe(*query, *doit, true, *all));
+    });
+}
+
+int dasi_free_wipe(const dasi_wipe_t* wipe) {
+    return tryCatch([wipe] {
+        ASSERT(wipe);
+        delete wipe;
+    });
+}
+
+int dasi_wipe_next(dasi_wipe_t* wipe) {
+    return tryCatch(std::function<int()> {[wipe] {
+        ASSERT(wipe);
+        if (wipe->first) {
+            wipe->first = false;
+        } else {
+            ++wipe->iterator;
+        }
+        if (wipe->iterator == wipe->generator.end()) { return DASI_ITERATION_COMPLETE; }
+        return DASI_SUCCESS;
+    }});
+}
+
+int dasi_wipe_attrs(const dasi_wipe_t* wipe, const char** value) {
```

Comment:
This feel like the wrong name for this. We aren't asking for (one of) many attributes of an iterator object. This clearly just iterates over string output.